cnd/src/btsieve/bitcoin/mod.rs

```diff
@@ -6,25 +6,30 @@ pub use self::{
     bitcoind_connector::BitcoindConnector, cache::Cache, transaction_ext::TransactionExt,
 };
 use crate::{
-    btsieve::{BlockByHash, LatestBlock, Predates},
-    transaction, Never,
+    btsieve::{find_relevant_blocks, BlockByHash, BlockHashes, LatestBlock, Predates},
+    transaction,
 };
 use bitcoin::{
     consensus::{encode::deserialize, Decodable},
     BitcoinHash, OutPoint,
 };
 use chrono::NaiveDateTime;
-use futures_core::compat::Future01CompatExt;
-use genawaiter::{
-    sync::{Co, Gen},
-    GeneratorState,
-};
+use genawaiter::{sync::Gen, GeneratorState};
 use reqwest::{Client, Url};
-use std::collections::HashSet;
 
 type Hash = bitcoin::BlockHash;
 type Block = bitcoin::Block;
 
+impl BlockHashes<Hash> for Block {
```

Comment:
I understand your motivation of putting these in one trait but it is not very idiomatic Rust to do that.

Traits are not interfaces that describe instances of objects.
They abstract over functionality and for that, a good rule of thumb is to have one trait per functionality.